src/kvstore/kvstore_dist.h

```diff
@@ -619,15 +629,17 @@ class KVStoreDist : public KVStoreLocal {
           push_pskv.keys.push_back(ps_key);
           pull_pskv.keys.push_back(ps_key);
           // push_pskv stores lengths of compressed blocks
-          push_pskv.lens.push_back(part_compr);
+          push_pskv.lens.push_back(part_compr * num_bytes);
           // pull_pskv stores lengths of original data
-          pull_pskv.lens.push_back(part_orig);
+          pull_pskv.lens.push_back(part_orig * num_bytes);
           push_pskv.size += part_compr;
           pull_pskv.size += part_orig;
         }
-        CHECK_EQ(static_cast<size_t>(push_pskv.size), compr_size);
-        CHECK_EQ(static_cast<size_t>(pull_pskv.size), original_size);
-        CHECK_EQ(push_pskv.lens.size(), num_servers*2);
+        push_pskv.size *= num_bytes;
+        pull_pskv.size *= num_bytes;
+        CHECK_EQ(static_cast<size_t>(push_pskv.size), compr_size * num_bytes);
+        CHECK_EQ(static_cast<size_t>(pull_pskv.size), original_size * num_bytes);
+        CHECK_EQ(push_pskv.lens.size(), num_servers * 2);
```

Comment:
Can we not do something like:
CHECK_EQ(static_cast<size_t>(push_pskv.size), compr_size);
CHECK_EQ(static_cast<size_t>(pull_pskv.size), original_size);
push_pskv.size *= num_bytes;
pull_pskv.size *= num_bytes;
instead so that we perform 2 less multiplications?

Reply:
Good idea, will do